src/components/PopupWithForm.js

```diff
@@ -8,26 +8,28 @@ export default class PopupWithForm extends Popup {
       this._popupForm.querySelectorAll(".modal__input")
     );
     this._handleFormSubmit = handleFormSubmit;
+   
   }
 
   _getInputValues() {
     const formValues = {};
     this._inputList.forEach((input) => {
       formValues[input.name] = input.value;
     });
+    console.log("Captured form values:", formValues);
     return formValues;
   }
 
   setEventListeners() {
     super.setEventListeners();
-
     this._popupForm.addEventListener("submit", (event) => {
-      event.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this._popupForm.reset();
-      this.close();
+        event.preventDefault();
+        const formData = this._getInputValues();
+        console.log("Form Data on Submit:", formData);
+        this._handleFormSubmit(formData);
+        this._popupForm.reset();
```

Comment:
You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.